sdk/search/azure-search/src/main/java/com/azure/search/AutocompletePagedResponse.java

```diff
@@ -3,29 +3,44 @@
 
 package com.azure.search;
 
-import com.azure.core.http.rest.Page;
 import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.http.rest.Response;
 import com.azure.core.http.rest.SimpleResponse;
 import com.azure.search.models.AutocompleteItem;
 import com.azure.search.models.AutocompleteResult;
 
 /**
- * Represents an HTTP response from the autocomplete API request that contains a list of items deserialized into a
- * {@link Page}.
+ * This class represents a response from the autocomplete API. It contains the {@link AutocompleteItem
+ * AutocompleteItems} returned from the service.
  */
-public class AutocompletePagedResponse extends PagedResponseBase<Void, AutocompleteItem> {
+public final class AutocompletePagedResponse extends PagedResponseBase<Void, AutocompleteItem> {
+    private final Double coverage;
 
     /**
-     * Constructor
+     * Creates an {@link AutocompletePagedResponse} from the returned {@link Response}.
      *
-     * @param autocompleteResponse an http response with the results
+     * @param autocompleteResponse Autocomplete response returned from the service.
      */
-    public AutocompletePagedResponse(SimpleResponse<AutocompleteResult> autocompleteResponse) {
+    AutocompletePagedResponse(SimpleResponse<AutocompleteResult> autocompleteResponse) {
         super(autocompleteResponse.getRequest(),
             autocompleteResponse.getStatusCode(),
             autocompleteResponse.getHeaders(),
             autocompleteResponse.getValue().getResults(),
             null,
             null);
+
+        this.coverage = autocompleteResponse.getValue().getCoverage();
```

Comment:
Can getValue() be null?

Reply:
Shouldn't, `null` value is likely an error scenario.

Reply:
What does a HTTP 204 resolve to? Would this have a non-null value?

Reply:
The autocomplete API only expects `200` as a valid response code.